File: scratch_fix_fonts.py
import os
import re

src_dir = r"c:\Users\MOHAMED\Downloads\food-donation-system\frontend\src"

# This regex matches `fontFamily: "..."` or `fontFamily: '...'` with optional preceding/trailing commas
# It handles spaces well.
pattern1 = re.compile(r',\s*fontFamily:\s*["\'][^"\']+["\']')
pattern2 = re.compile(r'fontFamily:\s*["\'][^"\']+["\']\s*,?')

def fix_fonts(directory):
    count = 0
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith('.jsx') or file.endswith('.js'):
                path = os.path.join(root, file)
                with open(path, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                new_content = pattern1.sub('', content)
                new_content = pattern2.sub('', new_content)
                
                if new_content != content:
                    with open(path, 'w', encoding='utf-8') as f:
                        f.write(new_content)
                    print(f"Updated {path}")
                    count += 1
    print(f"Total files updated: {count}")

if __name__ == "__main__":
    fix_fonts(src_dir)

File: scratch_fix_fonts.js
const fs = require('fs');
const path = require('path');

const srcDir = path.join('c:', 'Users', 'MOHAMED', 'Downloads', 'food-donation-system', 'frontend', 'src');

const pattern1 = /,\s*fontFamily:\s*["'][^"']+["']/g;
const pattern2 = /fontFamily:\s*["'][^"']+["']\s*,?/g;

let count = 0;

function walk(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);
        if (stat.isDirectory()) {
            walk(fullPath);
        } else if (file.endsWith('.jsx') || file.endsWith('.js')) {
            const content = fs.readFileSync(fullPath, 'utf8');
            let newContent = content.replace(pattern1, '');
            newContent = newContent.replace(pattern2, '');
            if (newContent !== content) {
                fs.writeFileSync(fullPath, newContent, 'utf8');
                console.log(`Updated ${fullPath}`);
                count++;
            }
        }
    }
}

walk(srcDir);
console.log(`Total files updated: ${count}`);
